### DISCUSSION TITLE
Can Coyote be used for other Deterministic Simulation Testing besides concurrency?

### DISCUSSION THREAD

**Discussion by NatElkins** [microsoft/coyote#514] [General]
There's been a lot of hype about Deterministic Simulation Testing, with projects like FoundationDB, TigerBeetle, Antithesis, etc making very effective use of it.  I found out about Coyote by reading an article that gives an overview of the landscape: https://databases.systems/posts/open-source-antithesis-p1

I am curious, can Coyote only be used for concurrency issues or can it also be used to inject things like network, disk, time, etc.  And if so, is there any documentation or examples?

Great project, thank you!

**Comment by akashlal** [comment]
Hi @NatElkins, 
Think of Coyote as increasing coverage for tests that exercise concurrency. It focuses on the exploration algorithm that gives you this bump in coverage. It doesn't have any native support for injecting network failures, etc. The test creation responsibility is on the programmer.

**Reply by NatElkins** [reply]
Got it, thank you!  I suppose I was wondering more on a theoretical basis.  For example, if I wanted to create a shim around disk access and model that with Coyote actors.  Have you ever seen any examples of using Coyote actors to take advantage of the DST capabilities of Coyote that didn't necessarily use Task as the underlying primitive/mechanism?

**Reply by akashlal** [reply]
Yes, indeed, see this for example: https://www.microsoft.com/en-us/research/uploads/prod/2021/09/poolmanager-coyote.pdf
